doc/nrf/libraries/others/profiler.rst

```diff
@@ -13,130 +13,196 @@ The output is provided using RTT and can be visualized in a custom Python backen
-* :c:func:`profiler_log_send` - Send profiled data.
+      * :c:func:`profiler_log_start` - Start logging.
+      * :c:func:`profiler_log_send` - Send profiled data.
 
-It is good practice to wrap the calls in one function that you then call to profile event occurrences.
-The following code example shows a function for profiling an event with data::
+   #. Add one or more of the following optional functions in-between the mandatory functions, depending on the data format:
 
-	static void profile_data_event(uint32_t val1, int32_t val2, int16_t val3,
-				       uint8_t val4, const char *string)
-	{
-		struct log_event_buf buf;
+      * :c:func:`profiler_log_encode_uint32` - Add 32-bit unsigned integer connected with the event.
+      * :c:func:`profiler_log_encode_int32` - Add 32-bit integer connected with the event.
+      * :c:func:`profiler_log_encode_uint16` - Add 16-bit unsigned integer connected with the event.
+      * :c:func:`profiler_log_encode_int16` - Add 16-bit integer connected with the event.
+      * :c:func:`profiler_log_encode_uint8` - Add 8-bit unsigned integer connected with the event.
+      * :c:func:`profiler_log_encode_int8` - Add 8-bit integer connected with the event.
+      * :c:func:`profiler_log_encode_string` - Add string connected with the event.
 
-		profiler_log_start(&buf);
-		/* Profiling data connected with an event */
-		profiler_log_encode_uint32(&buf, val1);
-		profiler_log_encode_int32(&buf, val2);
-		profiler_log_encode_int16(&buf, val3);
-		profiler_log_encode_uint8(&buf, val4);
-		profiler_log_encode_string(&buf, string);
-		profiler_log_send(&buf, data_event_id);
-	}
+#. Wrap the calls in one function that you then call to profile event occurrences.
+   The following code example shows a function for profiling an event with data:
 
-.. note::
+   .. code-block:: c
 
-	The event ID and the data that is profiled with the event must be consistent with the registered event type.
-	The data for every data field must be provided in the correct order.
+      static void profile_data_event(uint32_t val1, int32_t val2, int16_t val3,
+                  uint8_t val4, const char *string)
+      {
+        struct log_event_buf buf;
 
+        profiler_log_start(&buf);
+        /* Profiling data connected with an event */
+        profiler_log_encode_uint32(&buf, val1);
+        profiler_log_encode_int32(&buf, val2);
+        profiler_log_encode_int16(&buf, val3);
+        profiler_log_encode_uint8(&buf, val4);
+        profiler_log_encode_string(&buf, string);
+        profiler_log_send(&buf, data_event_id);
+      }
 
-Supported backends
-******************
+   .. note::
 
-The Profiler supports a custom backend to visualize the output data.
+	    The ``data_event_id`` and the data that is profiled with the event must be consistent with the registered event type.
+	    The data for every data field must be provided in the correct order.
+
+Configuring profiling for Event Manager events
```

Comment:
Isn't better to join this part with "Configuration for use with Event Manager" - it would be more clear for a user.

Reply:
Reworked, thanks.